ci-operator/config/openstack-k8s-operators/openstack-k8s-operators-ci/openstack-k8s-operators-openstack-k8s-operators-ci-main.yaml

```diff
@@ -49,6 +57,24 @@ images:
       as:
       - registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.18-openshift-4.12
   to: ci-build-root-golang-1.19-sdk-1.31
+- build_args:
+  - name: SDK_VERSION
+    value: v1.31.0
+  dockerfile_path: images/build-root/Dockerfile.rhel-9.custom
+  inputs:
+    rhel-9-golang-1.20:
+      as:
+      - registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.18-openshift-4.12
+  to: ci-build-root-golang-1.20-sdk-1.31
+- build_args:
+  - name: SDK_VERSION
+    value: v1.31.0
+  dockerfile_path: images/build-root/Dockerfile.rhel-9.custom
+  inputs:
+    rhel-9-golang-1.21:
+      as:
+      - registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.18-openshift-4.12
```

Comment:
sanity check please.

first - we are using this same image in all of these 

```
rhel-8-release-golang-1.18-openshift-4.12 
```

Also, we are now adding rhel-9 images but still consuming the rhel-8 openshift container is that OK

Reply:
Hi @marios, thanks for the review. 

This line is a little confusing but what we are doing is overriding [1], it's a method documented here[2] where we can use the same Dockerfile for multiple container releases.

This patch is just about getting the containers building and available for use. A follow up PR with be submitted this quater to move prow jobs over to the new rhel9, golang 1.21 container.

[1] https://github.com/openstack-k8s-operators/openstack-k8s-operators-ci/blob/1723abd4880521c07cd9e3a5b84f6e578afeb015/images/build-root/Dockerfile.custom#L1C1-L1C91
[2] https://docs.ci.openshift.org/docs/architecture/images/

Reply:
Yeah, but we are missing a detail here. Since we moved to a new Dockerfile (Dockerfile.rhel-9.custom), the `FROM X` is different there, so we must replace `registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.18-openshift-4.12` with `registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.20-openshift-4.14` [1] here too. See in the image build log for 1.21, it is still using the 1.20 image, since there was no replace [2].

[1] https://github.com/openstack-k8s-operators/openstack-k8s-operators-ci/blob/main/images/build-root/Dockerfile.rhel-9.custom#L1
[2] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/46906/rehearse-46906-pull-ci-openstack-k8s-operators-openstack-k8s-operators-ci-main-images/1742343451271237632/artifacts/build-logs/ci-build-root-golang-1.21-sdk-1.31-amd64.log

Reply:
Great! Let me quickly update this.